mosaic_api/mosaic_api/api/api_v1/endpoints/mosaics.py

```diff
@@ -0,0 +1,188 @@
+"""Mosaic endpoints."""
+from mosaic_api.models.mosaics import Mosaic, MosaicRequest, Link
+import random
+import string
+from fastapi import APIRouter, Response, HTTPException
+from starlette import status
+from starlette.requests import Request
+from cogeo_mosaic.mosaic import MosaicJSON
+from rio_tiler.io import COGReader
+from pystac_client import Client
+from typing import List, Optional
+from asyncio import wait_for, create_task
+import asyncio
+import aiohttp
+
+from mosaic_api.core.config import MOSAIC_API_ROOT
+
+router = APIRouter()
+
+
+@router.get(
+    "/mosaics/{mosaic_id}",
+    responses={
+        200: {"description": "return mosaic for a given ID"}
+    },
+    response_model=Mosaic,
+)
+async def get_mosaic(
+        mosaic_id: str
+) -> Mosaic:
+    """Return mosaic info for the given mosaic ID"""
+    # todo: should this check if the mosaic actually exists, or just return the tilejson url?
+    if mosaic_id:
+        return _mosaic(mosaic_id)
+    else:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="invalid mosaic ID")
```

Comment:
you can make path parameter required using `fastapi.Path(...)`  and also add description. As we do in https://github.com/developmentseed/titiler/blob/master/titiler/core/titiler/core/factory.py#L297

Reply:
you can also mix path/query definition using `Dependencies` 

e.g: this is the dependency we use in api.cogeo.xyz.

```python
@dataclass
class MosaicParams:
    """Create mosaic path from args"""

    layer: str = Query(..., description="Mosaic Layer name ('{username}.{layer}')")

    def __post_init__(self,):
        """Define mosaic URL."""
        pattern = (
            r"^(?P<username>[a-zA-Z0-9-_]{1,32})\.(?P<layername>[a-zA-Z0-9-_]{1,32})$"
        )
        match = re.match(pattern, self.layer)
        if not match:
            raise HTTPException(
                status_code=400, detail=f"Invalid layer name: `{self.layer}`",
            )

        meta = match.groupdict()
        self.username = meta["username"]
        self.layername = meta["layername"]
        if mosaic_config.backend == "dynamodb://":
            self.url = f"{mosaic_config.backend}{mosaic_config.host}:{self.layer}"
        else:
            self.url = f"{mosaic_config.backend}{mosaic_config.host}/{self.username}/{self.layername}{mosaic_config.format}"
```